File: algorithms/selection_sort/selection_sort.py
def selection_sort(arr):
    for i in range(len(arr) - 1):
        min_index = i
        for j in range(i, len(arr)):
            if (arr[j] < arr[min_index]):
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]

File: algorithms/selection_sort/selection_sort.rs
fn selection_sort<T: Ord + Copy + Clone>(arr: &mut [T]) {
    for i in 0..arr.len()-1 {
        let mut min_index = i.clone();
        for j in i..arr.len() {
            if arr[j] < arr[min_index] {
                min_index = j;
            }
        }
        arr.swap(i, min_index);
    }
}